java-symbolic-execution/src/test/java/org/sonar/java/se/xproc/ExceptionalYieldTest.java

```diff
@@ -138,11 +134,37 @@ void exceptional_yields_void_method() {
     assertThat(exceptionalYields).hasSize(3);
     assertThat(exceptionalYields.stream().filter(y -> y.exceptionType(semanticModel).isUnknown()).count()).isEqualTo(1);
 
-    MethodYield explicitExceptionYield = exceptionalYields.stream().filter(y -> y.exceptionType(semanticModel).is("org.foo.MyException1")).findAny().get();
-    assertThat(explicitExceptionYield.parametersConstraints.get(0).get(ObjectConstraint.class)).isEqualTo(ObjectConstraint.NULL);
+    ExceptionalYield explicit = exceptionalYields.stream().filter(y -> y.exceptionType(semanticModel).is("org.foo.MyException1")).findAny().get();
+    assertThat(paramObjectConstraint(explicit)).isEqualTo(ObjectConstraint.NULL);
 
-    MethodYield implicitExceptionYield = exceptionalYields.stream().filter(y -> y.exceptionType(semanticModel).is("org.foo.MyException2")).findAny().get();
-    assertThat(implicitExceptionYield.parametersConstraints.get(0).get(ObjectConstraint.class)).isEqualTo(ObjectConstraint.NOT_NULL);
+    ExceptionalYield implicit = exceptionalYields.stream().filter(y -> y.exceptionType(semanticModel).is("org.foo.MyException2")).findAny().get();
+    assertThat(paramObjectConstraint(implicit)).isEqualTo(ObjectConstraint.NOT_NULL);
   }
 
+  @Test
+  void exceptional_yields_anonymous_exception() {
+    Pair<SymbolicExecutionVisitor, Sema> sevAndSemantic = createSymbolicExecutionVisitorAndSemantic(
+      mainCodeSourcesPath("symbolicexecution/behaviorcache/AnonymousExceptionalYield.java"),
+      new NullDereferenceCheck());
+
+    MethodBehavior mb = getMethodBehavior(sevAndSemantic.a, "foo");
+    List<MethodYield> yields = mb.yields();
+    assertThat(yields).hasSize(3);
+
+    Sema semanticModel = sevAndSemantic.b;
+    List<ExceptionalYield> exceptionalYields = mb.exceptionalPathYields().collect(Collectors.toList());
+    assertThat(exceptionalYields).hasSize(2);
+    assertThat(exceptionalYields.stream().filter(y -> y.exceptionType(semanticModel).isUnknown()).count()).isEqualTo(1);
```

Comment:
I wonder if we really need to test all these here.
I am thinking that the priority is to test that the anonymous `RuntimeException` is handled well. I.e., that it is returned in the `exceptionalPathYields` list and that its `parameter constraint` is correct.

I agree it is a bit of nitpicking so I would not mind if you leave it as is.